pkg/apis/core/validation/validation.go

```diff
@@ -1154,6 +1154,71 @@ func validateProjectionSources(projection *core.ProjectedVolumeSource, projectio
 				allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
 			}
 		}
+		if projPath := srcPath.Child("clusterTrustBundlePEM"); source.ClusterTrustBundle != nil {
+			numSources++
+
+			usingName := source.ClusterTrustBundle.Name != nil
+			usingSignerName := source.ClusterTrustBundle.SignerName != nil
+
+			switch {
+			case usingName && usingSignerName:
+				allErrs = append(allErrs, field.Invalid(projPath, source.ClusterTrustBundle, "only one of name and signerName may be used"))
+			case usingName:
+				if *source.ClusterTrustBundle.Name == "" {
+					allErrs = append(allErrs, field.Required(projPath.Child("name"), "must be a valid object name"))
+				}
+
+				name := *source.ClusterTrustBundle.Name
+				if splitPoint := strings.LastIndex(name, ":"); splitPoint != -1 {
+					// This looks like it refers to a signerName trustbundle.
+					signerName := strings.ReplaceAll(name[:splitPoint], ":", "/")
+					if signerNameErrs := ValidateSignerName(projPath.Child("name"), signerName); len(signerNameErrs) != 0 {
+						allErrs = append(allErrs, field.Invalid(projPath.Child("name"), name, "not a valid clustertrustbundle name"))
+					}
+
+					additionalName := name[splitPoint+1:]
+					if errMsgs := apimachineryvalidation.NameIsDNSSubdomain(additionalName, false); len(errMsgs) != 0 {
+						allErrs = append(allErrs, field.Invalid(projPath.Child("name"), name, "not a valid clustertrustbundle name"))
+					}
+				} else {
+					// This refers to a non-signerName trustbundle
+					if errMsgs := apimachineryvalidation.NameIsDNSSubdomain(name, false); len(errMsgs) != 0 {
+						allErrs = append(allErrs, field.Invalid(projPath.Child("name"), name, "not a valid clustertrustbundle name"))
+					}
+				}
```

Comment:
it's good to check the name is well-formed, it's not great to separate this logic from `validateClusterTrustBundleName`

I would suggest:
1. move `validateClusterTrustBundleName` to core validation and export it
2. add a `ExtractSignerNameFromClusterTrustBundleName(string) (string, error)` function that does the `:` detection and escaping reversing

Use it here like:
```go
if signerName, err := ExtractSignerNameFromClusterTrustBundleName(name); err != nil {
  allErrs = append(allErrs, malformed bad signerName err)
} else {
  allErrs = append(allErrs, validateClusterTrustBundleName(name, signerName))
}
```